### PR TITLE
[Fix#534] IndexError / Worker initialization sets wrong last_file_index 

### LINKED ISSUE #534: IndexError / Worker initialization sets wrong last_file_index
**Describe the bug**

During initialization `Worker` [sets ](https://github.com/miurahr/py7zr/blob/a157a9c5a5c93bbf9e338b607c1fc2865479217b/py7zr/py7zr.py#L1223)instance field `self.last_file_index = len(self.files)` which is totally wrong for non-empty list of files. Lately when working with archive this bug raises *IndexError: list index out of range*.

Such behavior is caused by side-effects while working with `Worker` hence on second and further initializations it becomes corrupted.

**To Reproduce**

```python
import py7zr
with py7zr.SevenZipFile('test.7z', 'w') as archive:
	archive.writestr('1', '1.txt')
	archive.test()
```

> Traceback (most recent call last):
>   File "1.py", line 99, in <module>
>     with py7zr.SevenZipFile('test.7z', 'w') as archive:
>   File "C:\Python311\Lib\site-packages\py7zr\py7zr.py", line 417, in __exit__
>     self.close()
>   File "C:\Python311\Lib\site-packages\py7zr\py7zr.py", line 1110, in close
>     self._write_flush()
>   File "C:\Python311\Lib\site-packages\py7zr\py7zr.py", line 697, in _write_flush
>     self.worker.flush_archive(self.fp, folder)
>   File "C:\Python311\Lib\site-packages\py7zr\py7zr.py", line 1512, in flush_archive
>     if "maxsize" in self.header.files_info.files[self.last_file_index]:
>                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
> IndexError: list index out of range

**Expected behavior**

image

**Environment:**
 - OS:  Windows 10
 - Python 3.11.5
 - py7zr version: 0.20.6

**Current workaround**

Just create new wrapper per each archive operation:

```python
import py7zr
with py7zr.SevenZipFile('test.7z', 'w') as archive:
	archive.writestr('1', '1.txt')
with py7zr.SevenZipFile('test.7z', 'r') as archive:
	archive.test()
```

**Possible fixes**

1. Correctly [initialize worker](https://github.com/miurahr/py7zr/blob/a157a9c5a5c93bbf9e338b607c1fc2865479217b/py7zr/py7zr.py#L1223).

```python
self.last_file_index = len(self.files) - 1
```

2. Alternative bad solution is to prevent `Worker` reinitialization. On of many approaches:

```python
class SevenZipFile:
	def __init__(self, ...):
		self._worker = None
		...

	@property
	def worker(self):
		if not self._worker:
			raise NotInitializedError(...)
		return self._worker
	
	@worker.setter
	def worker(self, value):
		if self._worker:
			raise AlreadyInitializedError(...)
		self._worker = value
```

### PR DESCRIPTION
**Pull request by baterflyrity**
https://github.com/miurahr/py7zr/issues/534

### CODE REVIEW

**Comment by miurahr** [comment]
`pip -e check` failed 
```
  tests/test_basic.py:277:1: BLK100 Black would make changes.
  tests/test_basic.py:277:1: E302 expected 2 blank lines, found 1
```

recommended you to run `black` to format the code.

File: tests/test_basic.py [hunk 1]
```diff
@@ -273,3 +273,9 @@ def test_py7zr_list_values():
     assert file_list[1].crc32 == 0xB36AAEDB
     assert file_list[2].crc32 == 0xDCBF8D07
     assert file_list[3].crc32 == 0x80FC72BE
+
+@pytest.mark.basic
```
Comment (miurahr) [inline tests/test_basic.py:277]:
```suggestion


@pytest.mark.basic
```

**Comment by baterflyrity** [comment]
Thanks. Hope, this time it will pass.

**Comment by miurahr** [comment]
You can run `tox -e check`  and see errors on your desktop
```
tox: check
  check: commands[0]> flake8 py7zr tests
  tests/test_basic.py:281:26: BLK100 Black would make changes.
```

File: tests/test_basic.py [hunk 2]
```diff
@@ -273,3 +273,10 @@ def test_py7zr_list_values():
     assert file_list[1].crc32 == 0xB36AAEDB
     assert file_list[2].crc32 == 0xDCBF8D07
     assert file_list[3].crc32 == 0x80FC72BE
+
+
+@pytest.mark.basic
+def test_multiple_uses():
+    with py7zr.SevenZipFile(os.path.join(testdata_path, "test_multiple.7z"), "w") as archive:
+        archive.writestr('1', '1.txt')
```
Comment (miurahr) [inline tests/test_basic.py:281]: When apply `black` command, it changed
```suggestion
        archive.writestr("1", "1.txt")
```

**Pull request merged**